src/components/IdentityBadge.js

```diff
@@ -1,13 +1,50 @@
-import React from 'react'
+import React, { useCallback, useEffect, useState } from 'react'
+import { useHistory } from 'react-router-dom'
 import { IdentityBadge as Badge } from '@1hive/1hive-ui'
+
 import { getNetworkType } from '../lib/web3-utils'
+import { getProfileForAccount } from '../lib/profile'
 
-function IdentityBadge({ entity, ...props }) {
-  return <Badge entity={entity} networkType={getNetworkType()} {...props} />
-}
+const addressCache = new Map()
+
+function IdentityBadge({ entity, useBox, ...props }) {
+  const [profileName, setProfileName] = useState(null)
+
+  const history = useHistory()
+  const handleViewProfile = useCallback(() => {
+    history.push(`/profile?account=${entity}`)
+  }, [entity, history])
+
+  useEffect(() => {
+    let cancelled = false
+    async function fetchProfile() {
+      if (addressCache.get(entity)) {
+        setProfileName(addressCache.get(entity))
+        return
+      }
+
+      const profile = await getProfileForAccount(entity)
```

Comment:
do we have some possible errors here?

Reply:
It could error when fetching the public profile. I'll create an issue to work on general error handling so we can work on it after we have most of it merged!

Reply:
sounds good!